app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -560,6 +564,22 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
         }
     }
 
+    /**
+     * Try to migrate the topic specific engine to the first general or custom search engine available.
+     */
+    @Suppress("NestedBlockDepth")
+    private fun migrateTopicSpecificSearchEngines() {
+        components.core.store.state.search.selectedOrDefaultSearchEngine.let { currentSearchEngine ->
+            if (currentSearchEngine?.isGeneral == false) {
+                components.core.store.state.search.searchEngines.firstOrNull() { nextSearchEngine ->
+                    nextSearchEngine.isGeneral
+                }.let {
+                    if (it != null) components.useCases.searchUseCases.selectSearchEngine(it)
+                }
```

Comment:
The `let` scope function is used most of the time in combination with the safe call operator - `?.` to execute a lambda only on non-null objects:

```suggestion
                }?.let {
                   components.useCases.searchUseCases.selectSearchEngine(it)
                }
```